core/router/gateway/gateway.go

```diff
@@ -32,14 +33,19 @@ type Gateway struct {
 	Schedule    Schedule
 	LastSeen    time.Time
 
-	token string
+	mu            sync.RWMutex // Protect token and Authenticated
```

Comment:
either _token_ should also start with capital or _Authenticated_ should be lowercase